src/osd/scrubber/osd_scrub_sched.h

```diff
@@ -0,0 +1,346 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <atomic>
+#include <chrono>
+#include <memory>
+#include <optional>
+#include <vector>
+
+#include "common/RefCountedObj.h"
+#include "osd/osd_types.h"
+#include "osd/scrubber_common.h"
+
+#include "utime.h"
+
+class PG;
+
+namespace Scrub {
+
+// possible outcome when trying to select a PG and scrub it
+enum class attempt_t {
+  scrub_initiated,     // successfully started a scrub
+  none_ready,	       // no pg to scrub
+  no_local_resources,  // failure to secure local OSD scrub resource
+  already_started,     // already started scrubbing this pg
+  no_such_pg,	       // can't find this pg
+  bad_pg_state,	       // pg state (clean, active, etc.)
+  preconditions	       // time, configuration, etc.
```

Comment:
again, `preconditions` can be hardly be considered an attempt result .

Reply:
@tchaikov (wasn't making any fixes to this PR, as I was working on solving two issues detected when testing it. For one - I found out that it wasn't specific to this change, and issued a PR yesterday. And now I solved the other one.
Will be pushing new version)